src/redux/Books/books.js

```diff
@@ -1,44 +1,77 @@
+/* eslint-disable */
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import url from '../API/api';
+
 const ADD = 'bookstore/books/ADD';
 const REMOVE = 'bookstore/books/REMOVE';
+const GET_BOOKS = 'bookstore/books/GET_BOOKS';
+
+export const fetchThunks = () => async (dispatch) => {
+  const apiFetch = await fetch(url, {
+    method: 'GET',
+    body: JSON.stringify(),
+    headers: {
+      accept: 'application/json',
+    },
+  }).then((data) => data.json());
+  const firstState = [];
+  Object.keys(apiFetch).forEach((e) => {
+    firstState.push({ ...apiFetch[e][0], item_id: e });
+  });
+  dispatch({
+    type: GET_BOOKS,
+    payload: firstState,
+  });
+};
+
+// add new book
+export const addNewBook = (book) => async (dispatch) => {
+    await fetch( url, {
+      method: 'POST',
+      headers: {
+        'content-type': 'application/json',
+      },
+      body: JSON.stringify(book),
+    }).then(() => dispatch(fetchThunks()));
+}
+
+// export const addNewBook =  createAsyncThunk(
```

Comment:
Please be sure to use createAsyncThunk for creating thunk action creators so that the project meets the requirements 🧐 Also, kindly remove all the commented code from your repo so that its cleaner and professional 🚀